### ISSUE TITLE
Sinjid Room Transition Crash 

### ISSUE THREAD

**Issue by TomFulp** [ruffle-rs/ruffle#1942]
https://www.newgrounds.com/portal/view/209736/format/flash?emulate=flash

Start the game, select a character and you will begin in a room. Walk up to the door at the top and press spacebar to enter - you will get the following crash error:

# Error Info
Error name: Error
Error message: panicked at 'called `Option::unwrap()` on a `None` value', core\src\display_object\container.rs:779:18
Error stack:
```
Error: panicked at 'called `Option::unwrap()` on a `None` value', core\src\display_object\container.rs:779:18
    at Object.H (https://uploads.ungrounded.net/ruffle_wrapper/lib/core.ruffle.78b9038e746cb0a65e0c.js:1:20372)
    at __wbg_new_df6e6ab7a65c4c4d (https://uploads.ungrounded.net/ruffle_wrapper/lib/ruffle.js?1607530732:1:34984)
    at https://uploads.ungrounded.net/ruffle_wrapper/lib/d61c8cde6500241c6dc5.module.wasm:wasm-function[1057]:0x206eeb
    at https://uploads.ungrounded.net/ruffle_wrapper/lib/d61c8cde6500241c6dc5.module.wasm:wasm-function[1381]:0x24599f
    at https://uploads.ungrounded.net/ruffle_wrapper/lib/d61c8cde6500241c6dc5.module.wasm:wasm-function[3698]:0x2f9753
    at https://uploads.ungrounded.net/ruffle_wrapper/lib/d61c8cde6500241c6dc5.module.wasm:wasm-function[2925]:0x2dc0cc
    at https://uploads.ungrounded.net/ruffle_wrapper/lib/d61c8cde6500241c6dc5.module.wasm:wasm-function[3995]:0x2fe40d
    at https://uploads.ungrounded.net/ruffle_wrapper/lib/d61c8cde6500241c6dc5.module.wasm:wasm-function[4025]:0x2fe970
    at https://uploads.ungrounded.net/ruffle_wrapper/lib/d61c8cde6500241c6dc5.module.wasm:wasm-function[3900]:0x2fcfdd
    at https://uploads.ungrounded.net/ruffle_wrapper/lib/d61c8cde6500241c6dc5.module.wasm:wasm-function[4026]:0x2fe9a6
```

# Player Info
Allows script access: false

# Page Info
Page URL: https://uploads.ungrounded.net/ruffle_wrapper/ruffleembed.html?v=1.0.64&browsermode=default&props=%7B%22ruffle%22%3A%22%2Fruffle_wrapper%2Flib%2Fruffle.js%3F1607530732%22%2C%22public_path%22%3A%22%2Fruffle_wrapper%2Flib%2F%22%2C%22swf%22%3A%22https%3A%2F%2Fuploads.ungrounded.net%2F209000%2F209736_Sinjid2.swf%3F1104780350%22%2C%22vars%22%3A%7B%22NewgroundsAPI_PublisherID%22%3A1%2C%22NewgroundsAPI_SandboxID%22%3A%225fd4c6f9e18de%22%2C%22NewgroundsAPI_SessionID%22%3A%22qvAV7rOTX1gn9FbIy63F356a192b7913b04c54574d18c28d46e6395428abIwYd%22%2C%22NewgroundsAPI_UserName%22%3A%22TomFulp%22%2C%22NewgroundsAPI_UserID%22%3A1%2C%22ng_username%22%3A%22TomFulp%22%7D%2C%22width%22%3A600%2C%22height%22%3A470%2C%22icon%22%3A%22https%3A%2F%2Fimg.ngfiles.com%2Fdefaults%2Ficon-portal-xl.gif%3Ff1601395813%22%7D

# Browser Info
Useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36
OS: Win32

# Ruffle Info
Version: 0.1.0
Name: nightly 2020-12-08
Channel: nightly
Built: 2020-12-08T19:08:00.321Z
Commit: 021194875a2897a56d3dc961597e5f75b8369ec6

**Comment by Toad06** [comment]
There's no panic in nightly builds <= 2020-11-27, though unexpected things can be displayed on screen:
![sc1](https://user-images.githubusercontent.com/14274782/101987619-8b4ff980-3c95-11eb-9c28-d186c1a80f53.png)


Compiling with https://github.com/ruffle-rs/ruffle/commit/6e565d4da26bbbee15dde1f4129b81d713a8c615 shows none of these problems - just like #1671:
![sc2](https://user-images.githubusercontent.com/14274782/101987625-96a32500-3c95-11eb-8153-f722e63b85c8.png)

**Comment by Herschel** [comment]
Fixed by #2228
